src/Runner.Worker/Handlers/StepHost.cs

```diff
@@ -230,10 +219,6 @@ public async Task<string> DetermineNodeRuntimeVersion(IExecutionContext executio
                                             bool inheritConsoleHandler,
                                             CancellationToken cancellationToken)
         {
-            // make sure container exist.
-            ArgUtil.NotNull(Container, nameof(Container));
```

Comment:
Leave arg checks if runner is NOT using `containerHooks`?

Reply:
Yes, I think we should leave arg cheks in the case when the runner isn't using container hooks.